sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/HelperTests.java

```diff
@@ -204,4 +205,62 @@ public void pageListCustomDeserializer() throws IOException {
         assertEquals(2, pageList.getPageRange().size());
         assertEquals(1, pageList.getClearRange().size());
     }
+
+    // Tests that container names are properly URL decoded when retrieved from BlobUrlParts. Container names with special characters are not supported
+    // by the service, however, the names should still be encoded.
+    @Test
+    public void containerNameDecodingOnGet() {
+        BlobUrlParts parts = new BlobUrlParts().setScheme("http").setHost("host").setContainerName("my%20container");
+        assertEquals("my container", parts.getBlobContainerName());
+        // URL should retain the encoded form supplied by caller
+        assertTrue(parts.toUrl().toString().contains("my%20container"));
+    }
+
+    @Test
+    public void blobNameDecodingOnGet() {
+        BlobUrlParts parts = new BlobUrlParts().setScheme("http").setHost("host").setBlobName("my%20blob");
+        assertEquals("my blob", parts.getBlobName());
+        // URL should retain the encoded form supplied by caller
+        assertTrue(parts.toUrl().toString().contains("my%20blob"));
+    }
+
+    // Tests that blob names are automatically URL encoded when set in BlobUrlParts.
+    @Test
+    public void setBlobNameEncodedInURL() {
+        BlobUrlParts parts = new BlobUrlParts().setScheme("http").setHost("host").setBlobName("my blob");
+        assertEquals("my blob", parts.getBlobName());
+        String url = parts.toUrl().toString();
+        assertFalse(url.contains("my blob"));
+        assertTrue(url.contains("%20"));
+    }
+
+    // Tests that container names are automatically URL encoded when set in BlobUrlParts.
+    @Test
+    public void setContainerNameEncodedInURL() {
+        BlobUrlParts parts = new BlobUrlParts().setScheme("http").setHost("host").setContainerName("my container");
+        assertEquals("my container", parts.getBlobContainerName());
+        String url = parts.toUrl().toString();
+        assertFalse(url.contains("my container"));
+        assertTrue(url.contains("%20"));
```

Comment:
For these assertions, it would be good to include the full expected encoded container name `my%20container` to make sure we are explicitly encoding the container name